### ISSUE TITLE
segfault when src=dst in copy rules

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#2560]
```
$ cat dune
(rule (copy# foo.ml foo.ml))

(executable (name foo))
$ dune build foo.exe
Error: exception Sys_error("_build/default/foo.ml: No such file or directory")
Backtrace:
Raised by primitive operation at file "stdlib.ml", line 390, characters 28-54
Called from file "src/stdune/io.ml", line 87, characters 17-37
Called from file "src/action_exec.ml", line 94, characters 4-318
Called from file "src/build_system.ml", line 1503, characters 10-100
Called from file "src/fiber/fiber.ml", line 112, characters 7-12
Re-raised at file "src/stdune/exn.ml" (inlined), line 39, characters 38-65
Called from file "src/stdune/exn_with_backtrace.ml", line 17, characters 2-40
Called from file "src/fiber/fiber.ml", line 82, characters 8-15
Re-raised at file "src/stdune/exn.ml" (inlined), line 39, characters 38-65
Called from file "src/stdune/exn_with_backtrace.ml", line 17, characters 2-40
Called from file "src/fiber/fiber.ml", line 82, characters 8-15
Re-raised at file "src/stdune/exn.ml" (inlined), line 39, characters 38-65
Called from file "src/stdune/exn_with_backtrace.ml", line 17, characters 2-40
Called from file "src/fiber/fiber.ml", line 82, characters 8-15
Re-raised at file "src/stdune/exn.ml" (inlined), line 39, characters 38-65
Called from file "src/stdune/exn_with_backtrace.ml", line 17, characters 2-40
Called from file "src/fiber/fiber.ml", line 82, characters 8-15
Re-raised at file "src/stdune/exn.ml" (inlined), line 39, characters 38-65
Called from file "src/stdune/exn_with_backtrace.ml", line 17, characters 2-40
Called from file "src/fiber/fiber.ml", line 82, characters 8-15

I must not segfault.  Uncertainty is the mind-killer.  Exceptions are
the little-death that brings total obliteration.  I will fully express
my cases.  Execution will pass over me and through me.  And when it
has gone past, I will unwind the stack along its path.  Where the
cases are handled there will be nothing.  Only I will remain.
```

**Comment by diml** [comment]
That's not good indeed

**Comment by emillon** [comment]
Hi,

I was checking stack overflow bugs and this one has been fixed by #3485.

Output is now just:

    Error: _build/default/foo.ml: No such file or directory
